experimental/compactdecimal/src/provider.rs

```diff
@@ -0,0 +1,104 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+// Provider structs must be stable.
+#![allow(clippy::exhaustive_structs, clippy::exhaustive_enums)]
+// Suppress a warning on zerovec::makevarule.
+#![allow(missing_docs)]
+
+//! Data provider struct definitions for this ICU4X component.
+//!
+//! Read more about data providers: [`icu_provider`]
+
+use alloc::borrow::Cow;
+use icu_provider::{yoke, zerofrom};
+use zerovec::ZeroMap2d;
+
+/// Relative time format V1 data struct.
+
+#[icu_provider::data_struct(
+    LongCompactDecimalFormatDataV1Marker = "compactdecimal/long@1",
+    ShortCompactDecimalFormatDataV1Marker = "compactdecimal/short@1"
+)]
+#[derive(Debug, Clone, Default, PartialEq)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+#[cfg_attr(
+    feature = "datagen", 
+    derive(serde::Serialize, databake::Bake),
+    databake(path = icu_compactdecimal::provider)
+)]
+#[yoke(prove_covariance_manually)]
+pub struct CompactDecimalPatternDataV1<'data> {
+    /// A map keyed on log10 of the CLDR `type` attribute and the `cldr` count attribute.
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub patterns: ZeroMap2d<'data, i8, Count, PatternULE>,
+}
+
+/// A CLDR plural keyword, or the explicit value 1.
+/// See <https://www.unicode.org/reports/tr35/tr35-numbers.html#Language_Plural_Rules>.
+#[zerovec::make_ule(CountULE)]
+#[zerovec::derive(Debug)]
+#[derive(Copy, Clone, PartialOrd, Ord, PartialEq, Eq, Debug)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+#[cfg_attr(
+    feature = "datagen", 
+    derive(serde::Serialize, databake::Bake),
+    databake(path = icu_compactdecimal::provider)
+)]
+#[repr(u8)]
+pub enum Count {
```

Comment:
This enum should is representing the general `Count` for any plural rules, for example in currency or units, we will need `Count` too.

Therefore, I think `Count` implementation should be in some central place that we can use it, instead of writing this implementation for each componenet.

Reply:
Yeah; for now let's duplicate it but we should figure out a common solution. I don't want to over-engineer this given that we're waiting on icu_pattern which is intended to be the common solution (@zbraniecki)

Reply:
How is pattern related to this enum?

Reply:
The enum is input into the pattern selection, which I imagine may influence the design of the enum